enhancements/ingress/ingress-fault-detection.md

```diff
@@ -0,0 +1,174 @@
+---
+title: ingress-fault-detection
+authors:
+  - "@sgreene570"
+reviewers:
+  - "@danehans"
+  - "@frobware"
+  - "@knobunc"
+  - "@miciah"
+  - "@candita"
+  - "@rfredette"
+approvers:
+  - "@knobunc"
+  - "@smarterclayton"
+creation-date: 2020-08-17
+last-updated: 2020-08-17
+status: partially implemented
+see-also:
+  - "https://github.com/openshift/enhancements/pull/274"
+replaces:
+superseded-by:
+---
+
+# Ingress Fault Detection and Pinpointing
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement enriches the [Ingress Operator](https://github.com/openshift/cluster-ingress-opreator)
+by granting the operator the ability to automatically detect when ingress-related issues occur within a cluster.
+This enhancement also gives Cluster Admins the ability to pinpoint which specific part of the cluster Ingress "chain" is at fault when
+external traffic bound for application pods is unable to reach (or experiences difficulties en-route to) the correct application destination.
+
+## Motivation
+
+Cluster Ingress issues can greatly and rapidly impact the health of an OpenShift cluster. Cluster Administrators should be alerted to
+any potential ingress complications via the Ingress Operator as soon as they occur, as to help minimize unplanned application downtime.
+OpenShift cluster administrators should not first find out about ingress breakages when they inspect the status of a different critical
+cluster component, such as the console. A Cluster Administrator should also have tooling available to them to thoroughly diagnose and
+understand why routes on their cluster are not working.
+
+This enhancement also has benefits for internal OpenShift development.
+Often times when networking-related bugs are identified within a cluster, determining which networking sub-component
+(router, SDN, DNS, etc.) is responsible for the observed bug is difficult. Bug reports tend to spend lots of time bouncing between different teams
+as engineers are not sure which component(s) is legitimately at fault. Being able to quickly pinpoint Ingress related issues will expedite the bugfix process.
+
+### Goals
+
+1. Teach the [Ingress Operator](https://github.com/openshift/cluster-ingress-operator) to automatically detect cluster Ingress issues
+and alert cluster administrators when relevant problems are discovered.
+2. Provide a secondary tool that can be triggered at will by a Cluster Administrator or an OpenShift support engineer that spawns an Ingress "debug"
+container that quickly identifies and triages potential Ingress issues within the cluster.
```

Comment:
To be clear, the second goal is follow-up work to the work already done in 4.7, right?

Reply:
yup, I tagged that goal as post 4.7